### ISSUE TITLE
Parser error recovery throw AssertionError when parsing invalid annotation decl in `class-change` branch

### ISSUE THREAD

**Issue by rdhananjaya** [ballerina-platform/ballerina-lang#25322]
**Description:**
Test case: 
    AnnotationDeclarationTest. testSourceOnlyAnnotDeclWithoutSource

Stack trace:
```


java.lang.IllegalStateException: const-decl
	at io.ballerinalang.compiler.internal.parser.BallerinaParserErrorHandler.getNextRuleForIdentifier(BallerinaParserErrorHandler.java:4108)
	at io.ballerinalang.compiler.internal.parser.BallerinaParserErrorHandler.getNextRule(BallerinaParserErrorHandler.java:2360)
	at io.ballerinalang.compiler.internal.parser.AbstractParserErrorHandler.fixAndContinue(AbstractParserErrorHandler.java:386)
	at io.ballerinalang.compiler.internal.parser.AbstractParserErrorHandler.fixAndContinue(AbstractParserErrorHandler.java:328)
	at io.ballerinalang.compiler.internal.parser.BallerinaParserErrorHandler.seekMatch(BallerinaParserErrorHandler.java:1053)
	at io.ballerinalang.compiler.internal.parser.AbstractParserErrorHandler.seekMatchInSubTree(AbstractParserErrorHandler.java:199)
	at io.ballerinalang.compiler.internal.parser.AbstractParserErrorHandler.seekInAlternativesPaths(AbstractParserErrorHandler.java:248)
	at io.ballerinalang.compiler.internal.parser.BallerinaParserErrorHandler.seekMatchInAlternativePaths(BallerinaParserErrorHandler.java:1533)
	at io.ballerinalang.compiler.internal.parser.BallerinaParserErrorHandler.seekMatch(BallerinaParserErrorHandler.java:1042)
	at io.ballerinalang.compiler.internal.parser.AbstractParserErrorHandler.seekMatchInSubTree(AbstractParserErrorHandler.java:199)
	at io.ballerinalang.compiler.internal.parser.AbstractParserErrorHandler.seekInAlternativesPaths(AbstractParserErrorHandler.java:248)
	at io.ballerinalang.compiler.internal.parser.BallerinaParserErrorHandler.seekMatchInAlternativePaths(BallerinaParserErrorHandler.java:1533)
	at io.ballerinalang.compiler.internal.parser.BallerinaParserErrorHandler.seekMatch(BallerinaParserErrorHandler.java:1042)
	at io.ballerinalang.compiler.internal.parser.AbstractParserErrorHandler.seekMatchInSubTree(AbstractParserErrorHandler.java:199)
	at io.ballerinalang.compiler.internal.parser.AbstractParserErrorHandler.seekInAlternativesPaths(AbstractParserErrorHandler.java:248)
	at io.ballerinalang.compiler.internal.parser.BallerinaParserErrorHandler.seekMatchInAlternativePaths(BallerinaParserErrorHandler.java:1533)
	at io.ballerinalang.compiler.internal.parser.BallerinaParserErrorHandler.seekMatch(BallerinaParserErrorHandler.java:1042)
	at io.ballerinalang.compiler.internal.parser.AbstractParserErrorHandler.seekMatchInSubTree(AbstractParserErrorHandler.java:199)
	at io.ballerinalang.compiler.internal.parser.AbstractParserErrorHandler.seekInAlternativesPaths(AbstractParserErrorHandler.java:248)
	at io.ballerinalang.compiler.internal.parser.BallerinaParserErrorHandler.seekMatchInAlternativePaths(BallerinaParserErrorHandler.java:1533)
	at io.ballerinalang.compiler.internal.parser.BallerinaParserErrorHandler.seekMatch(BallerinaParserErrorHandler.java:1042)
	at io.ballerinalang.compiler.internal.parser.AbstractParserErrorHandler.seekMatchInSubTree(AbstractParserErrorHandler.java:199)
	at io.ballerinalang.compiler.internal.parser.AbstractParserErrorHandler.fixAndContinue(AbstractParserErrorHandler.java:387)
	at io.ballerinalang.compiler.internal.parser.AbstractParserErrorHandler.fixAndContinue(AbstractParserErrorHandler.java:328)
	at io.ballerinalang.compiler.internal.parser.BallerinaParserErrorHandler.seekMatch(BallerinaParserErrorHandler.java:1053)
	at io.ballerinalang.compiler.internal.parser.AbstractParserErrorHandler.seekMatchInSubTree(AbstractParserErrorHandler.java:199)
	at io.ballerinalang.compiler.internal.parser.AbstractParserErrorHandler.seekInAlternativesPaths(AbstractParserErrorHandler.java:248)
	at io.ballerinalang.compiler.internal.parser.BallerinaParserErrorHandler.seekMatchInAlternativePaths(BallerinaParserErrorHandler.java:1533)
	at io.ballerinalang.compiler.internal.parser.BallerinaParserErrorHandler.seekMatch(BallerinaParserErrorHandler.java:1042)
	at io.ballerinalang.compiler.internal.parser.AbstractParserErrorHandler.seekMatchInSubTree(AbstractParserErrorHandler.java:199)
	at io.ballerinalang.compiler.internal.parser.AbstractParserErrorHandler.fixAndContinue(AbstractParserErrorHandler.java:387)
	at io.ballerinalang.compiler.internal.parser.AbstractParserErrorHandler.fixAndContinue(AbstractParserErrorHandler.java:328)
	at io.ballerinalang.compiler.internal.parser.BallerinaParserErrorHandler.seekMatch(BallerinaParserErrorHandler.java:1053)
	at io.ballerinalang.compiler.internal.parser.AbstractParserErrorHandler.seekMatchInSubTree(AbstractParserErrorHandler.java:199)
	at io.ballerinalang.compiler.internal.parser.AbstractParserErrorHandler.fixAndContinue(AbstractParserErrorHandler.java:380)
	at io.ballerinalang.compiler.internal.parser.AbstractParserErrorHandler.fixAndContinue(AbstractParserErrorHandler.java:328)
	at io.ballerinalang.compiler.internal.parser.BallerinaParserErrorHandler.seekMatch(BallerinaParserErrorHandler.java:1053)
	at io.ballerinalang.compiler.internal.parser.AbstractParserErrorHandler.seekMatchInSubTree(AbstractParserErrorHandler.java:199)
	at io.ballerinalang.compiler.internal.parser.AbstractParserErrorHandler.seekMatch(AbstractParserErrorHandler.java:181)
	at io.ballerinalang.compiler.internal.parser.AbstractParserErrorHandler.recover(AbstractParserErrorHandler.java:102)
	at io.ballerinalang.compiler.internal.parser.AbstractParser.recover(AbstractParser.java:106)
	at io.ballerinalang.compiler.internal.parser.BallerinaParser.parseSourceKeyword(BallerinaParser.java:7126)
	at io.ballerinalang.compiler.internal.parser.BallerinaParser.parseAnnotationAttachPoint(BallerinaParser.java:7093)
	at io.ballerinalang.compiler.internal.parser.BallerinaParser.parseAnnotationAttachPoints(BallerinaParser.java:7015)
	at io.ballerinalang.compiler.internal.parser.BallerinaParser.parseAnnotationDeclAttachPoints(BallerinaParser.java:6957)
	at io.ballerinalang.compiler.internal.parser.BallerinaParser.parseAnnotationDeclWithOptionalType(BallerinaParser.java:6900)
	at io.ballerinalang.compiler.internal.parser.BallerinaParser.parseAnnotationDeclFromType(BallerinaParser.java:6849)
	at io.ballerinalang.compiler.internal.parser.BallerinaParser.parseAnnotationDeclaration(BallerinaParser.java:6815)
	at io.ballerinalang.compiler.internal.parser.BallerinaParser.parseConstDecl(BallerinaParser.java:6031)
	at io.ballerinalang.compiler.internal.parser.BallerinaParser.parseConstantDeclaration(BallerinaParser.java:6014)
	at io.ballerinalang.compiler.internal.parser.BallerinaParser.parseTopLevelNode(BallerinaParser.java:758)
	at io.ballerinalang.compiler.internal.parser.BallerinaParser.parseTopLevelNode(BallerinaParser.java:299)
	at io.ballerinalang.compiler.internal.parser.BallerinaParser.parseTopLevelNode(BallerinaParser.java:239)
	at io.ballerinalang.compiler.internal.parser.BallerinaParser.parseCompUnit(BallerinaParser.java:149)
	at io.ballerinalang.compiler.internal.parser.BallerinaParser.parse(BallerinaParser.java:94)
	at io.ballerinalang.compiler.syntax.tree.SyntaxTree.from(SyntaxTree.java:55)
```

**Steps to reproduce:**

**Affected Versions:**

**OS, DB, other environment details and versions:**

**Related Issues (optional):**

**Suggested Labels (optional):**

**Suggested Assignees (optional):**

**Comment by rdhananjaya** [comment]
This issue is fixed in class-change branch, need to close after merging it to the master

**Comment by rdhananjaya** [comment]
Fixed in #25637
